demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt

```diff
@@ -0,0 +1,417 @@
+            val index = i
+            textViews[index].setOnEditorActionListener { v, actionId, event ->
+                if (actionId == EditorInfo.IME_ACTION_NEXT ||
+                        actionId == EditorInfo.IME_ACTION_DONE ||
+                        actionId == EditorInfo.IME_NULL
+                                && event.action == KeyEvent.ACTION_DOWN
+                                && event.keyCode == KeyEvent.KEYCODE_ENTER) {
+                    if (index + 1 < textViews.size) {
+                        textViews[index + 1].requestFocus()
+                    } else if (index == textViews.size - 1) {
+                        val inputMethodManager = activity
+                                .getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+                        inputMethodManager.hideSoftInputFromWindow(v.windowToken, 0)
+                    }
+                }
+                true
+            }
+
+            // Suppress the key focus change by KeyEvent.ACTION_UP of the enter key
+            textViews[index].setOnKeyListener { _, keyCode, event -> keyCode == KeyEvent.KEYCODE_ENTER && event.action == KeyEvent.ACTION_UP }
+        }
+
+    }
+
+    private fun alignSelfAsString(alignSelf: Int): String {
+        when (alignSelf) {
+            AlignSelf.AUTO -> return ALIGN_SELF_AUTO
+            AlignItems.FLEX_START -> return ALIGN_SELF_FLEX_START
+            AlignItems.FLEX_END -> return ALIGN_SELF_FLEX_END
+            AlignItems.CENTER -> return ALIGN_SELF_CENTER
+            AlignItems.BASELINE -> return ALIGN_SELF_BASELINE
+            AlignItems.STRETCH -> return ALIGN_SELF_STRETCH
+            else -> return ALIGN_SELF_AUTO
+        }
+    }
+
+    private inner class FlexEditTextWatcher internal constructor(val context: Context,
+                                                                 val textInputLayout: TextInputLayout,
+                                                                 val inputValidator: InputValidator,
+                                                                 val errorMessageId: Int) : TextWatcher {
+
+        override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {
+            // No op
+        }
+
+        override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
+            if (inputValidator.isValidInput(s)) {
+                textInputLayout.isErrorEnabled = false
+                textInputLayout.error = ""
+            } else {
+                textInputLayout.isErrorEnabled = true
+                textInputLayout.error = activity.resources.getString(errorMessageId)
+            }
+        }
+
+        override fun afterTextChanged(editable: Editable) {
+            if (textInputLayout.isErrorEnabled || TextUtils.isEmpty(editable) ||
+                    !inputValidator.isValidInput(editable.toString())) {
+                return
+            }
+            val value: Number
```

Comment:
Couldn't find a good way of mixing Int and Float in a single variable.
Instead parsed as Float first, then cast to Int later if needed. Please let me know if there is better way